### PR TITLE
remove dependency on shared resources package

### LINKED ISSUE #138: Make Monex use JavaScript libs specified in package.json instead of /src/main/xar-resources/resources/scripts
Monex loads JavaScript resources from src/main/xar-resources/resources/scripts. This means although JQuery 3.5.x is specified in package.json it still uses JQuery 1.9.x from src/main/xar-resources/resources/scripts.

### PR DESCRIPTION
**Pull request by line-o**
closes #138

- load frontend dependencies from npm
- add gulpfile
- integrate npm and gulpfile in pom.xml
- replace dependencies to shared resources  

CHANGES:
Some frontend libraries were updated or replaced 
- update AdminLTE to v2.4.18 (this updated its dependencies like jQuery, bootstrap and many others)
- update d3 to v3.5.17
- datatables.net-responsive (and datatables.net-responsive-bs to v2.2.9
- update Knockout to v3.5.1
- update knockout.mapping to v2.4.3
- replace ACE editor with prismjs v1.23.0 as it was only used for code highlighting
- declare xquery version 3.1 in all modules
- view.xql serialization options do no longer enforce-xhtml

FIXES
- yank on collapse/open of sidebar
- javascript errors on index-keys.html and other subpages
- align jQuery versions (was 1.7.1 _and_ 1.9.1)
- refactor deprecated jQuery calls

### CODE REVIEW

**Comment by line-o** [comment]
I wonder if d3 is used anywhere. I could not find where it is used. Can someone shed light on this?
If we find it is really not necessary then it would be good to drop the dependency altogether.

**Comment by duncdrum** [comment]
@windauer i think used it in the tree viewer that he recently removed, maybe he knows if it was used anywhere else

**Comment by windauer** [comment]
yes, afaik @duncdrum is righ and I introduced it back then and forgot to remove it. So please remove it @line-o

**Comment by line-o** [comment]
Travis build needs latest LTS of nodeJS but https://github.com/eXist-db/monex/pull/155/commits/3d118e0aa55cd6f863944b509fa02483acbde531 didn't do it :(

**Comment by line-o** [comment]
build is back \o/

File: src/main/xar-resources/resources/scripts/loadsource.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                exide.focus();
+                setTimeout(function() {
+                    if ($.browser.msie ||
+                        (typeof exide.eXide.app.hasFocus == "function" && !exide.eXide.app.hasFocus())) {
+                        alert("Opened code in existing eXide window.");
+                    }
+                }, 200);
+            } else {
+                window.eXide_onload = function() {
+                    console.log("onloaed called");
```
Comment (duncdrum) [inline src/main/xar-resources/resources/scripts/loadsource.js:31]: typo "onload"

**Review by duncdrum — CHANGES_REQUESTED** [review]
one typo otherwise LGTM, tests pass

File: src/main/xar-resources/resources/scripts/loadsource.js (hunk 1 above)
Comment (line-o) [reply src/main/xar-resources/resources/scripts/loadsource.js:31]: I will remove this line.

**Review by duncdrum — APPROVED** [review]

**Pull request merged**